### ISSUE TITLE
New Resource: `aws_cloudfrontkeyvaluestore_key`

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#35793]
Following on from the `aws_cloudfront_key_value_store` resource, add a new resource `aws_cloudfrontkeyvaluestore_key` to manage an individual key (and associated value) in a store.

* [API Reference](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_Operations_Amazon_CloudFront_KeyValueStore.html)

```hcl
resource "aws_cloudfrontkeyvaluestore_key" "example" {
  key   = "k1"
  value = "v1"
}
```

Relates https://github.com/hashicorp/terraform-provider-aws/issues/34512.
Relates https://github.com/hashicorp/terraform-provider-aws/pull/35663.

**Comment by womblep** [comment]
@ewbankkit you would need to have a link to the store resource as well

**Comment by surenduhan** [comment]
I thought this was being provided with the original PR of key_value_store.

We need multiple things to fully support this feature.

1. Being able to create and update the resource i.e. KV-Store (key_value_store) . This is merged in 5.37.0
2. Need modification for cloudfront_function to support arn or id of this key_value_store to be associated with cloudfront_function resource. Otherwise old resource will remove association on update of function code.
3. This Key-Value pair to be created, modified, and removed. This PR tries to do something, for point 3, but ideal would be to take a JSON or YAML as input to create the list of key values. 

With any of the above not being done, this feature is still not useful for cloud automation.
